apps/cli-daemon/src/app/workspace/workspace.service.ts

```diff
@@ -0,0 +1,87 @@
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
+import {
+  createWorkspaceHost,
+  // writeWorkspace,
+  readWorkspace as devKitReadWorkspace,
+  ProjectDefinition,
+  TargetDefinition,
+  WorkspaceDefinition,
+} from '@angular-devkit/core/src/workspace';
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common';
+
+import { SessionService } from '../session/session.service';
+
+export const BAD_PATH_EXCEPTION =
+  'Unable to determine format for workspace path.';
+export const NOT_ANGULAR_WORKSPACE =
+  'Unable to locate a workspace file for workspace path. Are you missing an `angular.json`' +
+  ' or `.angular.json` file?';
+
+@Injectable()
+export class WorkspaceService {
+  private readonly logger = new Logger(WorkspaceService.name);
+
+  constructor(private readonly sessionService: SessionService) {}
+
+  async readWorkspace(path: string): Promise<WorkspaceDefinition> {
+    try {
+      return (
+        await devKitReadWorkspace(
+          path,
+          createWorkspaceHost(new NodeJsSyncHost())
+        )
+      ).workspace;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
```

Comment:
please set a type

Reply:
There is no type, it's the error coming from the catch.

Reply:
you can use Error but I see TS wants it to be "unknown" so maybe "unknown" and remove the eslint-disable

Reply:
If it will be unknown, I won't be able to read its message property